dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -210,17 +210,18 @@ await foreach (Completions completions in response)
         CancellationToken cancellationToken)
     {
         var result = new List<ReadOnlyMemory<float>>(data.Count);
-        foreach (string text in data)
+
+        if (data.Count > 0)
         {
-            var options = new EmbeddingsOptions(this.DeploymentOrModelName, new[] { text });
+            var response = await RunRequestAsync(() => this.Client.GetEmbeddingsAsync(new(this.DeploymentOrModelName, data), cancellationToken)).ConfigureAwait(false);
+            var embeddings = response.Value.Data;
 
-            Response<Azure.AI.OpenAI.Embeddings> response = await RunRequestAsync(() => this.Client.GetEmbeddingsAsync(options, cancellationToken)).ConfigureAwait(false);
-            if (response.Value.Data.Count == 0)
+            if (embeddings.Count != data.Count)
             {
-                throw new KernelException("Text embedding not found");
+                throw new KernelException($"Expected {data.Count} text embedding(s), but received {embeddings.Count}");
             }
 
-            result.Add(response.Value.Data[0].Embedding.ToArray());
+            result.AddRange(embeddings.OrderBy((x) => x.Index).Select((x) => x.Embedding));
```

Comment:
Is there a special reason why we are applying the order by index in here? I suggest not mutating the original payload and get the embedding arrays in the provided order by the server if there's no real benefit to it.

Reply:
It's probably not necessary but, considering the index is a part of the response and I see no explicit language in the API reference that guarantees the response will come back ordered, a max 16 element sort operation seems like a small price to pay. Happy to remove if there's a different opinion or some clarification around response ordering guarantees?

Reply:
@anthonypuppo I think we can avoid ordering on SK side. If that's needed, users can always perform ordering on their side.

Reply:
Updated to remove the ordering.

Reply:
Thank you!